### PR TITLE
Check if related items are published before publishing an article

### PR DESCRIPTION
**Pull request by pablopunk**
SDESK-3464
SDESK-3550

### CODE REVIEW

File: scripts/apps/authoring/authoring/directives/AuthoringDirective.js [hunk 1]
```diff
@@ -593,13 +595,33 @@ export function AuthoringDirective(superdesk, superdeskFlags, authoringWorkspace
...
-                        ok && performPublish();
-                    });
+                    }).then((ok) => ok ? performPublish() : false);
+
                     return;
                 }
 
-                return performPublish();
+                // Check if there's unpublished related items
+                return familyService.fetchRelatedItems($scope.item)
```
[the remark this answers is not in the document]
Comment (pablopunk) [reply scripts/apps/authoring/authoring/directives/AuthoringDirective.js]: @mdhaman if the definition is changed then `fetchRelatedItems` implementation should also change right?

[the remark this answers is not in the document]
Comment (mdhaman) [reply scripts/apps/authoring/authoring/directives/AuthoringDirective.js]: of course @pablopunk will wait for your change in implementation of `fetchRelatedItems`.

[the remark this answers is not in the document]
Comment (pablopunk) [reply scripts/apps/authoring/authoring/directives/AuthoringDirective.js]: @mdhaman but I think that should be a separate task, it's still related to epic SDESK-3461 but I think out of the scope for this task which is only validation on publishing

[the remark this answers is not in the document]
Comment (petrjasek) [reply scripts/apps/authoring/authoring/directives/AuthoringDirective.js]: I think this is different, related items from family service are duplicates of the same item, but here you're checking items which were set as related by users, so it should use something else to get it.

[the remark this answers is not in the document]
Comment (pablopunk) [reply scripts/apps/authoring/authoring/directives/AuthoringDirective.js]: @petrjasek you mean there's already another way to fetch related items or I should implement the new way of getting related items and using it here?

[the remark this answers is not in the document]
Comment (petrjasek) [reply scripts/apps/authoring/authoring/directives/AuthoringDirective.js]: you should create a new one, afaik we don't have it yet. and it shouldn't be part of family service imo, maybe on content

[the remark this answers is not in the document]
Comment (petrjasek) [reply scripts/apps/authoring/authoring/directives/AuthoringDirective.js]: yep

**Pull request merged**